lib/ahoy.rb

```diff
@@ -112,6 +116,23 @@ def self.instance
   def self.instance=(value)
     Thread.current[:ahoy] = value
   end
+
+  # @return [RequestStore] Request store
+  def self.request_store
+    RequestStore.store
+  end
+
+  # allow cookies variable based on the current request
+  def self.cookies
+    return server_side_cookies if request_store[:cookies].nil?
+
+    request_store[:cookies]
```

Comment:
Could you use `request_store[:cookies].presence || server_side_cookies`

Reply:
tsk tsk tsk. can't use a Rails method in a gem buddyo ;)

Reply:
You know what, I thought to myself, bet he's done it cos its a gem and presence is almost certainly in activesupport. Then I thought, yeah but maybe its still available to us

Reply:
no, I can't, because that means if you specifically set `Ahoy.cookies = false` for a request, it will use the default value of `Ahoy.server_side_cookies`